apps/github-api-integration/behaviours/fetch-repos.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const Model = require('../models/github');
+const _ = require('lodash');
+
+// Behaviour wrapped in a function here
+// allows us to pass in a config
+module.exports = config => {
+
+  return superclass => class extends superclass {
+
+    // `getValues` is part of the GET pipeline. For details
+    // see https://github.com/UKHomeOfficeForms/hof-behaviour-hooks.
+    getValues(req, res, next) {
+
+      // Get the user entered input previously
+      // stored on the session model
+      const name = req.sessionModel.get('name');
+
+      // Define an instance of the Model
+      // with attributes and config (uri)
+      const model = new Model({name}, config);
+
+      // Call `fetch` to get the resource
+      // from the url we configured
+      model.fetch().then(result => {
+
+          // Create a subset of the result
+          const repos = _.map(result, repo => ({
+            name: repo.name,
+            desc: repo.description,
+            url: repo.html_url
+          }));
```

Comment:
I'd probably move this chunk of code into a `parse` method on the model, so that data fetching and processing is encapsulated in the data model.